core/src/main/java/org/apache/iceberg/TableMetadataParser.java

```diff
@@ -226,8 +225,7 @@ public static void toJson(TableMetadata metadata, JsonGenerator generator) throw
       }
     }
 
-    if (metadata.rowLineageEnabled()) {
-      generator.writeBooleanField(ROW_LINEAGE, metadata.rowLineageEnabled());
+    if (metadata.formatVersion() >= 3) {
```

Comment:
There is a constant in TableMetadata for the 3 here, just a nit but I think it's easier to see if (version > row lineage) do row lineage stuff